watertap/unit_models/gac.py

```diff
@@ -760,209 +708,303 @@ def eq_dg(b, t, j):
             )
 
-        @self.Constraint(doc="Length of the mass transfer zone")
-        def eq_length_mtz(b):
-            return b.length_mtz_replace == b.velocity_sup * b.ebct_mtz_replace
-
+        # mass balance for port not in control volume
         @self.Constraint(
-            doc="Length of the mass transfer zone at the time of replacement"
+            self.flowsheet().config.time,
+            self.target_species,
+            doc="mass balance for port not in control volume",
         )
-        def eq_approx_saturation(b):
-            return (1 * (b.bed_length - b.length_mtz_replace)) + (
-                ((b.saturation_mtz_upstream + b.conc_ratio_replace) / 2)
-                * b.length_mtz_replace
-            ) == b.bed_length * b.gac_saturation_replace
+        def eq_mass_transfer_port(b, t, j):
+            return (
+                b.gac_removed[t].get_material_flow_terms("Liq", j)
+                == -b.process_flow.mass_transfer_term[t, "Liq", j]
+            )
+
+        # no mass transfer of inert_species, fix to 0 to avoid 0 solver tolerances
+        for j in self.inert_species:
+            self.process_flow.mass_transfer_term[:, "Liq", j].fix(0)
+            self.gac_removed[0].get_material_flow_terms("Liq", j).fix(0)
 
         # ---------------------------------------------------------------------
         if (
             self.config.film_transfer_coefficient_type
             == FilmTransferCoefficientType.calculated
         ):
-            # TODO check N_Re formulation for packed beds
+
             self.N_Re = Var(
                 initialize=1,
                 bounds=(0, None),
                 domain=NonNegativeReals,
                 units=pyunits.dimensionless,
-                doc="Reynolds number, correlation using Schmidt number valid in Re < 2e4",
+                doc="Reynolds number, correlations using Reynolds number valid in Re < 2e4",
             )
-
             self.N_Sc = Var(
                 initialize=1000,
                 bounds=(0, None),
                 domain=NonNegativeReals,
                 units=pyunits.dimensionless,
-                doc="Schmidt number, correlation using Schmidt number valid in 0.7 < Sc < 1e4",
+                doc="Schmidt number, correlations using Schmidt number valid in 0.7 < Sc < 1e4",
             )
-
-            self.sphericity = Var(
+            self.shape_correction_factor = Var(
                 initialize=1,
                 bounds=(0, None),
                 domain=NonNegativeReals,
                 units=pyunits.dimensionless,
-                doc="Sphericity of the particle",
+                doc="shape correction factor",
             )
 
-            @self.Constraint(doc="Reynolds number calculation")
-            def eq_reynolds_number(b):
+            @self.Constraint(
+                self.flowsheet().config.time,
+                doc="Reynolds number calculation",
+            )
+            def eq_reynolds_number(b, t):
```

Comment:
I noticed a slight variation for calculating Re number in another paper by Crittenden et al., 1987. 
![image](https://user-images.githubusercontent.com/22691844/225881927-c8c68899-fa9d-41f6-9b02-ad19cc3033bd.png)
I think they are equivalent but just want to make sure. Your formulation includes particle diameter instead of 2*R, and you don't divide by bed voidage since that is accounted for by interstitial velocity (superficial velocity divided by bed voidage). Is that correct?